components/builder-events/src/event.rs

```diff
@@ -53,3 +53,22 @@ impl BuilderEvent {
 
     pub fn routing_key(self) -> Option<String> { self.routing_key }
 }
+
+#[macro_export]
+macro_rules! publish_event {
+    ($state:expr, $type:expr, $json:expr) => {
+        publish_event!($state, $type, $json, None)
+    };
+    ($state:expr, $type:expr, $json:expr, $routing_key:expr) => {
+        if let Some(bus) = $state.eventbus.as_ref() {
+            let event_type = $type;
+            match BuilderEvent::new(event_type, $routing_key, $json) {
+                Ok(event) => bus.inner.send(event).await,
```

Comment:
This `send` should be a method of `bus`, rather than requiring whatever's passed in to have a public `inner` member.

Reply:
Agreed. I handled this in https://github.com/habitat-sh/builder/pull/1540/commits/89ec48ec0e3da015458150f63ec464ee8a65a72c